course/src/isometry.cc

```diff
@@ -0,0 +1,115 @@
+#include <isometry.h>
+
+namespace ekumen {
+  namespace math {
+
+  const Vector3 Vector3::kUnitX(1.,0,0);
+  const Vector3 Vector3::kUnitY(0,1.,0);
+  const Vector3 Vector3::kUnitZ(0,0,1.);
+  const Vector3 Vector3::kZero(0,0,0);
+
+  Vector3 Vector3::operator + (const Vector3 & obj) const
+  {
+    return Vector3( x_ + obj.x(), y_ + obj.y(),z_ +  obj.z());
+  }
+
+  Vector3 Vector3::operator - (const Vector3 & obj) const
+  {
+    return Vector3( x_ - obj.x(), y_ - obj.y(),z_ -  obj.z());
+  }
+
+  bool Vector3::operator == (const std::initializer_list<double> & obj) const
+  {
+    if (obj.size() != 3) return false;
+    int i = 0;
+    for (auto & e : obj)
+    {
+      if (*data_[i] != e) return false;
+      i++;
+    }
+    return true;
+  }
+
+  bool Vector3::operator == (const Vector3 & obj) const
+  {
+    if (obj.x() != x_ || obj.y() != y_ || obj.z() != z_) return false;
```

Comment:
You should never compare floating point variables for equality. There are [multiple errors](https://www.embeddeduse.com/2019/08/26/qt-compare-two-floats/) that can come out of it. Use this: https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon